gpt2.py

```diff
@@ -89,9 +89,9 @@ def generate(inputs, params, n_head, n_tokens_to_generate):
     for _ in tqdm(range(n_tokens_to_generate), "generating"):  # auto-regressive decode loop
         logits = gpt2(inputs, **params, n_head=n_head)  # model forward pass
         next_id = np.argmax(logits[-1])  # greedy sampling
-        inputs = np.append(inputs, [next_id])  # append prediction to input
+        inputs = np.append(inputs, next_id)  # append prediction to input
 
-    return list(inputs[len(inputs) - n_tokens_to_generate :])  # only return generated ids
+    return [int(x) for x in inputs[-n_tokens_to_generate:]]  # only return generated ids
```

Comment:
Think this would be better so we can leave `inputs` as type list (actually I probably should've done this from the beginning since it's simpler):

```python
    inputs.append(int(next_id)) 
return inputs[len(inputs) - n_tokens_to_generate :]
```

Reply:
This means we also no longer need the `np.array(encoder.encode(prompt))` line

Reply:
Also, the same change should be made to `gpt2_pico.py`

Reply:
Good point, implemented in 4ebe614d2f9848ffe514a020c3f907530b623ca7.